src/Entity/Adherent.php

```diff
@@ -616,6 +628,23 @@ public function setCoordinatorManagedArea(CoordinatorManagedArea $coordinatorMan
         $this->coordinatorManagedArea = $coordinatorManagedArea;
     }
 
+    public function getBoardMember(): ?BoardMember
+    {
+        return $this->boardMember;
+    }
+
+    public function setBoardMember(string $area, ArrayCollection $roles): void
+    {
+        $this->boardMember = new BoardMember();
+        $this->boardMember->setArea($area);
+        $this->boardMember->setRoles($roles);
+    }
+
+    public function isBoardMember(): bool
+    {
+        return $this->boardMember instanceof BoardMember && !empty($this->boardMember->getArea() && !empty($this->boardMember->getRoles()));
```

Comment:
`&& !empty($this->boardMember->getArea()) && !empty($this->boardMember->getRoles())`